src/Pos/Explorer/Web/Server.hs

```diff
@@ -407,29 +435,32 @@ topsortTxsOrFail f =
     maybeThrow (Internal "Dependency loop in txs set") .
     topsortTxs f
 
+-- | Convert server address to client address,
+-- with the possible exception (effect).
 cAddrToAddr :: MonadThrow m => CAddress -> m Address
-cAddrToAddr cAddr =
-    fromCAddress cAddr &
-    either (const $ throwM $ Internal "Invalid address!") pure
+cAddrToAddr cAddr = either exception pure (fromCAddress cAddr)
```

Comment:
[Style guide](https://github.com/serokell/serokell-util/blob/master/serokell-style.md#point-free-style) only suggests avoiding over-using point-free style (which is not the case with `either exception pure . fromCAddress`). I think both are acceptable here depending on author's preferences.